crypto/evp/bio_ok.c

```diff
@@ -483,9 +483,11 @@ static int sig_in(BIO *b)
     void *md_data;
 
     ctx = BIO_get_data(b);
-    md = ctx->md;
+    if ((md = ctx->md) == NULL)
```

Comment:
not really necessary since both EVP_MD_CTX_get0_md() and EVP_MD_get_size() handle NULL

Reply:
I know, I figured the code was clearer with this check.  md is also passed to EVP_DigestInit_ex, EVP_DigestUpdate and EVP_DigestFinal_ex which don't handle NULL.

BTW: it never gets to the call to EVP_MD_CTX_get0_md_data which kind of validates the clarity improvement.